[TEST-FAILURE] MapNearCacheInvalidationFromClientTest Failures

```
java.lang.AssertionError: expected:<DefaultData{type=-7, hashCode=768969306, partitionHash=768969306, totalSize=9, dataSize=4, heapCost=29}> but was:<null>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.client.map.MapNearCacheInvalidationFromClientTest.testRemove(MapNearCacheInvalidationFromClientTest.java:208)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x/lastCompletedBuild/com.hazelcast$hazelcast-client/testReport/com.hazelcast.client.map/MapNearCacheInvalidationFromClientTest/testRemove/

```
java.lang.AssertionError: expected:<HeapData{type=-7, hashCode=768969306, partitionHash=768969306, totalSize=12, dataSize=4, heapCost=32}> but was:<null>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.client.map.MapNearCacheInvalidationFromClientTest.testRemove(MapNearCacheInvalidationFromClientTest.java:232)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK8-Quality-Outreach/com.hazelcast$hazelcast-client-legacy/228/testReport/junit/com.hazelcast.client.map/MapNearCacheInvalidationFromClientTest/testRemove/

**MapNearCacheInvalidationFromClientTest.testEvict**

```
java.lang.AssertionError: lite member near cache size should be less than 99 after evict but was 99
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.client.map.impl.nearcache.MapNearCacheInvalidationFromClientTest$3.run(MapNearCacheInvalidationFromClientTest.java:166)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:825)
	at com.hazelcast.client.map.impl.nearcache.MapNearCacheInvalidationFromClientTest.testEvict(MapNearCacheInvalidationFromClientTest.java:161)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/lab-fast-pr/143/testReport/junit/com.hazelcast.client.map.impl.nearcache/MapNearCacheInvalidationFromClientTest/testEvict/

--- a/hazelcast-client-legacy/src/test/java/com/hazelcast/client/map/impl/nearcache/MapNearCacheInvalidationFromClientTest.java
+++ b/hazelcast-client-legacy/src/test/java/com/hazelcast/client/map/impl/nearcache/MapNearCacheInvalidationFromClientTest.java
@@ -1,7 +1,6 @@
 package com.hazelcast.client.map.impl.nearcache;
 
 import com.hazelcast.cache.impl.nearcache.NearCache;
-import com.hazelcast.client.config.ClientConfig;
 import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.config.Config;
 import com.hazelcast.config.NearCacheConfig;
@@ -88,14 +87,18 @@ public class MapNearCacheInvalidationFromClientTest {
         }
 
         final IMap<Object, Object> liteMap = lite.getMap(mapName);
-
-        for (int i = 0; i < count; i++) {
-            assertNotNull(liteMap.get(i));
-        }
-
         final NearCache nearCache = getNearCache(lite, mapName);
-        final int sizeAfterPut = nearCache.size();
-        assertTrue("lite member near cache size should be > 0 but was " + sizeAfterPut, sizeAfterPut > 0);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                for (int i = 0; i < count; i++) {
+                    liteMap.get(i);
+                }
+                assertEquals(count, nearCache.size());
+            }
+        });
 
         map.clear();
 
@@ -103,10 +106,9 @@ public class MapNearCacheInvalidationFromClientTest {
             @Override
             public void run()
                     throws Exception {
-                final int sizeAfterClear = nearCache.size();
-                assertEquals("lite member near cache size should be 0 after clear but was " + sizeAfterClear, 0, sizeAfterClear);
+                assertEquals(0, nearCache.size());
             }
-        }, 10);
+        });
     }
 
     @Test
@@ -119,14 +121,18 @@ public class MapNearCacheInvalidationFromClientTest {
         }
 
         final IMap<Object, Object> liteMap = lite.getMap(mapName);
-
-        for (int i = 0; i < count; i++) {
-            assertNotNull(liteMap.get(i));
-        }
-
         final NearCache nearCache = getNearCache(lite, mapName);
-        final int sizeAfterPut = nearCache.size();
-        assertTrue("lite member near cache size should be > 0 but was " + sizeAfterPut, sizeAfterPut > 0);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                for (int i = 0; i < count; i++) {
+                    liteMap.get(i);
+                }
+                assertEquals(count, nearCache.size());
+            }
+        });
 
         map.evictAll();
 
@@ -134,10 +140,9 @@ public class MapNearCacheInvalidationFromClientTest {
             @Override
             public void run()
                     throws Exception {
-                final int sizeAfterEvict = nearCache.size();
-                assertEquals("lite member near cache size should be 0 after evict but was " + sizeAfterEvict, 0, sizeAfterEvict);
+                assertTrue(nearCache.size() < count);
             }
-        }, 10);
+        });
     }
 
     @Test
@@ -150,14 +155,18 @@ public class MapNearCacheInvalidationFromClientTest {
         }
 
         final IMap<Object, Object> liteMap = lite.getMap(mapName);
-
-        for (int i = 0; i < count; i++) {
-            assertNotNull(liteMap.get(i));
-        }
-
         final NearCache nearCache = getNearCache(lite, mapName);
-        final int sizeAfterPut = nearCache.size();
-        assertTrue("lite member near cache size should be > 0 but was " + sizeAfterPut, sizeAfterPut > 0);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                for (int i = 0; i < count; i++) {
+                    liteMap.get(i);
+                }
+                assertEquals(count, nearCache.size());
+            }
+        });
 
         map.evict(0);
 
@@ -165,11 +174,9 @@ public class MapNearCacheInvalidationFromClientTest {
             @Override
             public void run()
                     throws Exception {
-                final int sizeAfterEvict = nearCache.size();
-                assertTrue("lite member near cache size should be less than " + sizeAfterPut + " after evict but was "
-                        + sizeAfterEvict, sizeAfterEvict < sizeAfterPut);
+                assertTrue(nearCache.size() < count);
             }
-        }, 10);
+        });
     }
 
     @Test

--- a/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/MapNearCacheInvalidationFromClientTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/map/impl/nearcache/MapNearCacheInvalidationFromClientTest.java
@@ -1,7 +1,6 @@
 package com.hazelcast.client.map.impl.nearcache;
 
 import com.hazelcast.cache.impl.nearcache.NearCache;
-import com.hazelcast.client.config.ClientConfig;
 import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.config.Config;
 import com.hazelcast.config.NearCacheConfig;
@@ -88,14 +87,18 @@ public class MapNearCacheInvalidationFromClientTest {
         }
 
         final IMap<Object, Object> liteMap = lite.getMap(mapName);
-
-        for (int i = 0; i < count; i++) {
-            assertNotNull(liteMap.get(i));
-        }
-
         final NearCache nearCache = getNearCache(lite, mapName);
-        final int sizeAfterPut = nearCache.size();
-        assertTrue("lite member near cache size should be > 0 but was " + sizeAfterPut, sizeAfterPut > 0);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                for (int i = 0; i < count; i++) {
+                    liteMap.get(i);
+                }
+                assertEquals(count, nearCache.size());
+            }
+        });
 
         map.clear();
 
@@ -103,10 +106,9 @@ public class MapNearCacheInvalidationFromClientTest {
             @Override
             public void run()
                     throws Exception {
-                final int sizeAfterClear = nearCache.size();
-                assertEquals("lite member near cache size should be 0 after clear but was " + sizeAfterClear, 0, sizeAfterClear);
+                assertEquals(0, nearCache.size());
             }
-        }, 10);
+        });
     }
 
     @Test
@@ -119,14 +121,18 @@ public class MapNearCacheInvalidationFromClientTest {
         }
 
         final IMap<Object, Object> liteMap = lite.getMap(mapName);
-
-        for (int i = 0; i < count; i++) {
-            assertNotNull(liteMap.get(i));
-        }
-
         final NearCache nearCache = getNearCache(lite, mapName);
-        final int sizeAfterPut = nearCache.size();
-        assertTrue("lite member near cache size should be > 0 but was " + sizeAfterPut, sizeAfterPut > 0);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                for (int i = 0; i < count; i++) {
+                    liteMap.get(i);
+                }
+                assertEquals(count, nearCache.size());
+            }
+        });
 
         map.evictAll();
 
@@ -134,10 +140,9 @@ public class MapNearCacheInvalidationFromClientTest {
             @Override
             public void run()
                     throws Exception {
-                final int sizeAfterEvict = nearCache.size();
-                assertEquals("lite member near cache size should be 0 after evict but was " + sizeAfterEvict, 0, sizeAfterEvict);
+                assertTrue(nearCache.size() < count);
             }
-        }, 10);
+        });
     }
 
     @Test
@@ -150,14 +155,18 @@ public class MapNearCacheInvalidationFromClientTest {
         }
 
         final IMap<Object, Object> liteMap = lite.getMap(mapName);
-
-        for (int i = 0; i < count; i++) {
-            assertNotNull(liteMap.get(i));
-        }
-
         final NearCache nearCache = getNearCache(lite, mapName);
-        final int sizeAfterPut = nearCache.size();
-        assertTrue("lite member near cache size should be > 0 but was " + sizeAfterPut, sizeAfterPut > 0);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                for (int i = 0; i < count; i++) {
+                    liteMap.get(i);
+                }
+                assertEquals(count, nearCache.size());
+            }
+        });
 
         map.evict(0);
 
@@ -165,11 +174,9 @@ public class MapNearCacheInvalidationFromClientTest {
             @Override
             public void run()
                     throws Exception {
-                final int sizeAfterEvict = nearCache.size();
-                assertTrue("lite member near cache size should be less than " + sizeAfterPut + " after evict but was "
-                        + sizeAfterEvict, sizeAfterEvict < sizeAfterPut);
+                assertTrue(nearCache.size() < count);
             }
-        }, 10);
+        });
     }
 
     @Test

